### PR TITLE
Story4

### PR DESCRIPTION
**Pull request by Landwaterfood**
As a: Project Manager
I want: To be able to create new issues
So That: I can effectively manage projects

Requirements:

    Must follow this API spec:
    https://github.com/iO-Academy/IT-bug-tracking-API-template/blob/main/README.md#create-a-new-issue

### CODE REVIEW

File: report.php [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+$json = file_get_contents('php://input');
+
+$data = json_decode($json, true);
+
+$requiredExist = ValidationService::checkRequiredDataExists($data);
+
+if (!$requiredExist) {
+    http_response_code(400);
+    echo json_encode(["message" => "Invalid issue data"]);  // make sure to use the proper message fromt he API spec
+    return;  // could also use exit() to ensure an early exit
```
Comment (mallowmew) [inline report.php]: You should remove these comments

File: report.php [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+$passedValidation =
+    $data['title']
+    && $data['name']
+    && $severityExists
+    && $departmentExists;
+
+if (!$passedValidation) {
+    http_response_code(400);
+    echo json_encode(["message" => "Invalid issue data"]);
+    return;  // could also use exit()
```
Comment (mallowmew) [inline report.php]: remove comment

File: report.php [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+
+
+
+
+
+
+
+
+
```
Comment (mallowmew) [inline report.php]: Remove excess newlines

File: src/Entities/Comment.php [hunk 4]
```diff
@@ -8,4 +8,5 @@ class Comment
     public string $comment;
     public string $date_created;
     public int $issue_id;
+
```
Comment (mallowmew) [inline src/Entities/Comment.php:14]: Mysteriously appearing newline in entity

File: src/Entities/Issue.php [hunk 5]
```diff
@@ -19,14 +20,17 @@ public function getCompleted(): bool
         return $this->completed == 1;
     }
 
+
+
```
Comment (mallowmew) [inline src/Entities/Issue.php]: Excess newlines appeared?

File: src/Hydrators/IssueHydrator.php [hunk 6]
```diff
@@ -48,16 +48,50 @@ public static function getIssue($db, $issue_id)
...
         $commentsQuery->execute(['id' => $issue->id]);
         $commentsQuery->setFetchMode(PDO::FETCH_CLASS, Comment::class);
         $comments = $commentsQuery->fetchAll();
 
+        foreach($comments as $comment) {
+            $date = new \DateTime($comment->date_created);
+            $converted_date = $date->format("d/m/Y H:i");
+            $comment->date_created = $converted_date;
+        }
+
```
Comment (mallowmew) [inline src/Hydrators/IssueHydrator.php]: You should keep the date stored in the entity as flexible as possible until the point that it is required to be displayed.

When is the date "displayed" in the comment? How do we control the way that it is displayed?

File: src/Services/DateFormatter.php [hunk 7]
```diff
@@ -0,0 +1,11 @@
+<?php
+
+namespace ITBugTracking\Services;
+
+class DateFormatter
+{
+    public static function formatDate($date_created) {
+        $date = new \DateTime($date_created);
+        return $date->format("d/m/Y");
+    }
```
Comment (mallowmew) [inline src/Services/DateFormatter.php]: There are at least two date format functions...

**Review by mallowmew — CHANGES_REQUESTED** [review]

File: report.php (hunk 1 above)
Comment (Landwaterfood) [reply report.php]: done

File: report.php (hunk 2 above)
Comment (Landwaterfood) [reply report.php]: done

File: report.php (hunk 3 above)
Comment (Landwaterfood) [reply report.php]: done

File: src/Entities/Comment.php (hunk 4 above)
Comment (Landwaterfood) [reply src/Entities/Comment.php:14]: done

File: src/Entities/Issue.php (hunk 5 above)
Comment (Landwaterfood) [reply src/Entities/Issue.php]: done

**Review by mallowmew — APPROVED** [review]

**Pull request merged**